From 48d1e97b2b8c5853e35dea2391b44ae00625b780 Mon Sep 17 00:00:00 2001
From: Chris Lumens <clumens@redhat.com>
Date: Wed, 11 Aug 2010 10:38:13 -0400
Subject: [PATCH 1/6] Update systemd's default.target for the desired runlevel (#623102, mschmidt).

---
 pyanaconda/desktop.py |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/pyanaconda/desktop.py b/pyanaconda/desktop.py
index 49b7bbe..306e481 100644
--- a/pyanaconda/desktop.py
+++ b/pyanaconda/desktop.py
@@ -19,7 +19,7 @@
 # Author(s): Matt Wilson <msw@redhat.com>
 #
 
-import string
+import string, os
 from simpleconfig import SimpleConfigFile
 
 import logging
@@ -65,6 +65,14 @@ class Desktop (SimpleConfigFile):
             inittab.write (line)
         inittab.close ()
 
+        if not os.path.isdir(instPath + '/etc/systemd/system'):
+            log.warning("there is no /etc/systemd/system directory, cannot update default.target!")
+            return
+        default_target = instPath + '/etc/systemd/system/default.target'
+        if os.path.islink(default_target):
+            os.unlink(default_target)
+        os.symlink('/etc/systemd/system/runlevel' + str(self.runlevel) + '.target', default_target)
+
         if self.getDefaultDesktop():
             f = open(instPath + "/etc/sysconfig/desktop", "w")
             f.write(str (self))
-- 
1.7.1.1

