From 16ab88e311b726f4502a07bcd13609264fe985e0 Mon Sep 17 00:00:00 2001
From: Brian C. Lane <bcl@redhat.com>
Date: Mon, 06 May 2013 16:18:58 +0000
Subject: hostname has dropped -v option

---
diff --git a/pyanaconda/network.py b/pyanaconda/network.py
index 090141c..7487f32 100644
--- a/pyanaconda/network.py
+++ b/pyanaconda/network.py
@@ -672,7 +672,7 @@ def set_hostname(hn):
         return
 
     log.info("setting installation environment hostname to %s" % hn)
-    iutil.execWithRedirect("hostname", ["-v", hn ])
+    iutil.execWithRedirect("hostname", [hn])
 
 def write_hostname(rootpath, ksdata, overwrite=False):
     cfgfile = os.path.normpath(rootpath + hostnameFile)
--
cgit v0.9.1
