From a69b18013fefa55b0302a1a8404d08b46bb15968 Mon Sep 17 00:00:00 2001
From: Martin Sivak <msivak@redhat.com>
Date: Wed, 17 Oct 2012 09:14:14 +0000
Subject: Add missing import (#867296)

---
diff --git a/pyanaconda/ui/gui/spokes/source.py b/pyanaconda/ui/gui/spokes/source.py
index 61ce3ae..6fa7e8e 100644
--- a/pyanaconda/ui/gui/spokes/source.py
+++ b/pyanaconda/ui/gui/spokes/source.py
@@ -38,6 +38,7 @@ from pyanaconda.ui.gui.spokes import NormalSpoke
 from pyanaconda.ui.gui.categories.software import SoftwareCategory
 from pyanaconda.ui.gui.utils import enlightbox, gtk_thread_wait
 from pyanaconda.iutil import ProxyString, ProxyStringError
+from pyanaconda.ui.gui.utils import gtk_call_once
 
 __all__ = ["SourceSpoke"]
 
--
cgit v0.9.0.3
