diff -urN anaconda-22.20.3.orig/data/liveinst/liveinst anaconda-22.20.3/data/liveinst/liveinst
--- anaconda-22.20.3.orig/data/liveinst/liveinst	2015-03-05 23:19:00.748850936 +0300
+++ anaconda-22.20.3/data/liveinst/liveinst	2015-03-05 23:20:29.688316199 +0300
@@ -66,7 +66,10 @@
 # load modules that would get loaded by the initramfs (#230945)
 for i in raid0 raid1 raid5 raid6 raid456 raid10 dm-mod dm-zero dm-mirror dm-snapshot dm-multipath dm-round-robin vfat dm-crypt cbc sha256 lrw xts iscsi_tcp iscsi_ibft; do /sbin/modprobe $i 2>/dev/null ; done
 
-if [ -f /etc/system-release ]; then
+if [ -f /etc/rfremix-release ]; then
+    export ANACONDA_PRODUCTNAME=$( cat /etc/rfremix-release | sed -r -e 's/ *release.*//' )
+    export ANACONDA_PRODUCTVERSION=$( cat /etc/rfremix-release | sed -r -e 's/^.* ([0-9\.]+).*$/\1/' )
+elif [ -f /etc/system-release ]; then
     export ANACONDA_PRODUCTNAME=$( cat /etc/system-release | sed -r -e 's/ *release.*//' )
     export ANACONDA_PRODUCTVERSION=$( cat /etc/system-release | sed -r -e 's/^.* ([0-9\.]+).*$/\1/' )
 fi
@@ -77,7 +80,7 @@
 
 export ANACONDA_BUGURL=${ANACONDA_BUGURL:="https://bugzilla.redhat.com/bugzilla/"}
 
-RELEASE=$(rpm -q --qf '%{Release}' --whatprovides system-release)
+RELEASE=$(rpm -q --qf '%{Release}' --whatprovides rfremix-release)
 if [ "${RELEASE:0:2}" = "0." ]; then
     export ANACONDA_ISFINAL="false"
 else
diff -urN anaconda-22.20.3.orig/pyanaconda/bootloader.py anaconda-22.20.3/pyanaconda/bootloader.py
--- anaconda-22.20.3.orig/pyanaconda/bootloader.py	2015-03-05 23:19:00.745850987 +0300
+++ anaconda-22.20.3/pyanaconda/bootloader.py	2015-03-05 23:20:55.575869483 +0300
@@ -1491,7 +1491,7 @@
         defaults_file = "%s%s" % (iutil.getSysroot(), self.defaults_file)
         defaults = open(defaults_file, "w+")
         defaults.write("GRUB_TIMEOUT=%d\n" % self.timeout)
-        defaults.write("GRUB_DISTRIBUTOR=\"$(sed 's, release .*$,,g' /etc/system-release)\"\n")
+        defaults.write("GRUB_DISTRIBUTOR=\"$(test -f /etc/rfremix-release && sed 's, release .*$,,g' /etc/rfremix-release || sed 's, release .*$,,g' /etc/system-release)\"\n")
         defaults.write("GRUB_DEFAULT=saved\n")
         defaults.write("GRUB_DISABLE_SUBMENU=true\n")
         if self.console and self.has_serial_console:
