diff -urN anaconda-20.25.2.orig/pyanaconda/packaging/yumpayload.py anaconda-20.25.2/pyanaconda/packaging/yumpayload.py
--- anaconda-20.25.2.orig/pyanaconda/packaging/yumpayload.py	2013-10-22 12:25:09.752582566 +0400
+++ anaconda-20.25.2/pyanaconda/packaging/yumpayload.py	2013-10-22 12:26:06.061586733 +0400
@@ -84,7 +84,7 @@
 from pykickstart.constants import GROUP_ALL, GROUP_DEFAULT, KS_MISSING_IGNORE
 
 YUM_PLUGINS = ["blacklist", "whiteout", "fastestmirror", "langpacks"]
-default_repos = [productName.lower(), "rawhide"]
+default_repos = ["fedora", "rawhide"]
 
 import inspect
 import threading
