diff -urN anaconda-13.21.117.orig/scripts/upd-instroot anaconda-13.21.117/scripts/upd-instroot
--- anaconda-13.21.117.orig/scripts/upd-instroot	2011-10-10 11:32:09.182828631 +0400
+++ anaconda-13.21.117/scripts/upd-instroot	2011-10-11 13:40:45.147458686 +0400
@@ -1134,7 +1134,113 @@
 mv $DEST/usr/lib/anaconda-runtime/lib* $DEST/usr/$LIBDIR 2>/dev/null
 mv $DEST/usr/lib/anaconda/sitecustomize.py $DEST/usr/$LIBDIR/python?.?/site-packages
 
-mv $DEST/etc/yum.repos.d $DEST/etc/anaconda.repos.d
+rm -f $DEST/etc/yum.repos.d/*
+cat > $DEST/etc/yum.repos.d/russianfedoraremix-release-anaconda.repo <<EOF
+[reremix]
+name=RERemix \$releasever - \$basearch
+failovermethod=priority
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-reremix-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=0
+
+[reremix-security]
+name=RERemix \$releasever - \$basearch - Updates
+failovermethod=priority
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-reremix-security-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=0
+
+[reremix-fastbug]
+name=RERemix \$releasever - \$basearch - FastBug
+failovermethod=priority
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-reremix-fastbug-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=0
+
+#[rpmfusion-free-russianfedora]
+#name=RPM Fusion for Fedora \$releasever - Free
+#failovermethod=priority
+#mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-rpmfusion-free-\$releasever&arch=\$basearch
+#enabled=1
+#gpgcheck=0
+
+#[rpmfusion-free-reremix-updates]
+#name=RPM Fusion for Fedora \$releasever - Free - Updates
+#failovermethod=priority
+#mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-rpmfusion-free-updates-\$releasever&arch=\$basearch
+#enabled=1
+#gpgcheck=0
+
+#[rpmfusion-nonfree-russianfedora]
+#name=RPM Fusion for Fedora \$releasever - Non-Free
+#failovermethod=priority
+#mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-rpmfusion-nonfree-\$releasever&arch=\$basearch
+#enabled=1
+#gpgcheck=0
+
+#[rpmfusion-nonfree-reremix-updates]
+#name=RPM Fusion for Fedora \$releasever - Non-Free - Updates
+#failovermethod=priority
+#mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-rpmfusion-nonfree-updates-\$releasever&arch=\$basearch
+#enabled=1
+#gpgcheck=0
+
+[dag]
+name DAG for EL \$releasever
+failovermethod=priority
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=build-dag-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=0
+
+[epel]
+name=Extra Packages for Enterprise Linux 6 - \$basearch
+mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=epel-6&arch=\$basearch
+failovermethod=priority
+enabled=1
+gpgcheck=0
+
+[russianfedora-fixes]
+name=Russian Fedora for EL \$releasever - Fixes
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=fixes-reremix-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-reremix-fixes-fedora
+
+[russianfedora-fixes-updates]
+name=Russian Fedora for EL \$releasever - Fixes - Updates
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=fixes-reremix-updates-released-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-reremix-fixes-fedora
+
+[russianfedora-free]
+name=Russian Fedora for EL \$releasever - Free
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=free-reremix-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-reremix-free-fedora
+
+[russianfedora-free-updates]
+name=Russian Fedora for EL \$releasever - Free - Updates
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=free-reremix-updates-released-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-reremix-free-fedora
+
+[russianfedora-nonfree]
+name=Russian Fedora for EL \$releasever - Non-Free 
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=nonfree-reremix-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-reremix-nonfree-fedora
+
+[russianfedora-nonfree-updates]
+name=Russian Fedora for EL \$releasever - Non-Free - Updates
+mirrorlist=http://mirrors.rfremix.ru/mirrorlist?repo=nonfree-reremix-updates-released-\$releasever&arch=\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-reremix-nonfree-fedora
+EOF
 
 rm -f $DEST/usr/$LIBDIR/libunicode-lite*
 
