diff -urN anaconda-15.24.orig/data/bootdisk/i386/syslinux.cfg anaconda-15.24/data/bootdisk/i386/syslinux.cfg
--- anaconda-15.24.orig/data/bootdisk/i386/syslinux.cfg	2011-03-22 13:30:14.104505063 +0300
+++ anaconda-15.24/data/bootdisk/i386/syslinux.cfg	2011-03-22 13:31:34.627306687 +0300
@@ -24,6 +24,22 @@
   menu label Install system with ^basic video driver
   kernel vmlinuz
   append initrd=initrd.img xdriver=vesa nomodeset
+label quick-gnome
+  menu label Quick install GNOME (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/gnome.cfg
+label quick-kde
+  menu label Quick install KDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/kde.cfg
+label quick-xfce
+  menu label Quick install XFCE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/xfce.cfg
+label quick-lxde
+  menu label Quick install LXDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/lxde.cfg
 label rescue
   menu label ^Rescue installed system
   kernel vmlinuz
diff -urN anaconda-15.24.orig/data/bootdisk/x86_64/syslinux.cfg anaconda-15.24/data/bootdisk/x86_64/syslinux.cfg
--- anaconda-15.24.orig/data/bootdisk/x86_64/syslinux.cfg	2011-03-22 13:30:14.105505060 +0300
+++ anaconda-15.24/data/bootdisk/x86_64/syslinux.cfg	2011-03-22 13:31:56.722307702 +0300
@@ -24,6 +24,22 @@
   menu label Install system with ^basic video driver
   kernel vmlinuz
   append initrd=initrd.img xdriver=vesa nomodeset
+label quick-gnome
+  menu label Quick install GNOME (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/gnome.cfg
+label quick-kde
+  menu label Quick install KDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/kde.cfg
+label quick-xfce
+  menu label Quick install XFCE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/xfce.cfg
+label quick-lxde
+  menu label Quick install LXDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/lxde.cfg
 label rescue
   menu label ^Rescue installed system
   kernel vmlinuz
