diff -up anaconda-11.4.0.83/autopart.py.BAD anaconda-11.4.0.83/autopart.py
--- anaconda-11.4.0.83/autopart.py.BAD	2009-02-02 15:18:26.000000000 -0500
+++ anaconda-11.4.0.83/autopart.py	2009-02-02 15:18:46.000000000 -0500
@@ -195,10 +195,11 @@ class partlist:
         self.parts = []
 
 def getMinimumSector(disk):
-    if disk.type.name == 'sun':
-        start = long(disk.dev.sectors * disk.dev.heads)
-        start /= (1024 / disk.dev.sector_size)
-        return start + 1
+    # This is more trouble than it is worth.
+    # if disk.type.name == 'sun':
+    #    start = long(disk.dev.sectors * disk.dev.heads)
+    #    start /= (1024 / disk.dev.sector_size)
+    #    return start + 1
     return 0L
 
 # first step of partitioning voodoo
