diff -urN anaconda-15.29.orig/po/ru.po anaconda-15.29/po/ru.po
--- anaconda-15.29.orig/po/ru.po	2011-04-26 18:57:28.834821731 +0400
+++ anaconda-15.29/po/ru.po	2011-04-26 19:03:13.526010218 +0400
@@ -2143,19 +2143,43 @@
 "обычных задач работы с Интернетом. Поддержку каких дополнительных задач вы "
 "хотели бы включить?"
 
-#: pyanaconda/installclasses/fedora.py:48
-msgid "Graphical Desktop"
-msgstr "Графический рабочий стол"
+#: installclasses/fedora.py:48
+msgid "GNOME Desktop"
+msgstr "Рабочий стол GNOME"
+
+#: installclasses/fedora.py:53
+msgid "GNOME Desktop (Minimal)"
+msgstr "Рабочий стол GNOME (минимально)"
+
+#: installclasses/fedora.py:56
+msgid "KDE Desktop"
+msgstr "Рабочий стол KDE"
+
+#: installclasses/fedora.py:61
+msgid "KDE Desktop (Minimal)"
+msgstr "Рабочий стол KDE (минимально)"
+
+#: installclasses/fedora.py:64
+msgid "XFCE Desktop"
+msgstr "Рабочий стол XFCE"
+
+#: installclasses/fedora.py:69
+msgid "LXDE Desktop"
+msgstr "Рабочий стол LXDE"
+
+#: pyanaconda/installclasses/fedora.py:74
+msgid "Enlightenment Windows Manager"
+msgstr "оконный менеджер Enlightenment"
 
-#: pyanaconda/installclasses/fedora.py:53
+#: pyanaconda/installclasses/fedora.py:79
 msgid "Software Development"
 msgstr "Разработка ПО"
 
-#: pyanaconda/installclasses/fedora.py:59
+#: pyanaconda/installclasses/fedora.py:85
 msgid "Web Server"
 msgstr "Веб-сервер"
 
-#: pyanaconda/installclasses/fedora.py:63 pyanaconda/installclasses/rhel.py:44
+#: pyanaconda/installclasses/fedora.py:89 pyanaconda/installclasses/rhel.py:44
 msgid "Minimal"
 msgstr "Минимальный"
 
diff -urN anaconda-15.29.orig/pyanaconda/installclasses/fedora.py anaconda-15.29/pyanaconda/installclasses/fedora.py
--- anaconda-15.29.orig/pyanaconda/installclasses/fedora.py	2011-04-26 18:57:28.934823529 +0400
+++ anaconda-15.29/pyanaconda/installclasses/fedora.py	2011-04-26 18:59:53.224415484 +0400
@@ -34,8 +34,8 @@
 
 class InstallClass(BaseInstallClass):
     # name has underscore used for mnemonics, strip if you dont need it
-    id = "fedora"
-    name = N_("_Fedora")
+    id = "russianfedora"
+    name = N_("_RFRemix")
     _description = N_("The default installation of %s includes a set of "
                       "software applicable for general internet usage. "
                       "You can optionally select a different set of software "
@@ -45,11 +45,37 @@
     if productName.startswith("Red Hat Enterprise"):
         hidden = 1
 
-    tasks = [(N_("Graphical Desktop"),
+    tasks = [(N_("GNOME Desktop"),
               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
                "games", "gnome-desktop", "graphical-internet", "graphics",
                "hardware-support", "input-methods", "java", "office",
                "printing", "sound-and-video", "text-internet"]),
+             (N_("GNOME Desktop (Minimal)"),
+              ["base-x", "core", "gnome-desktop-minimal",
+               "hardware-support"]),
+             (N_("KDE Desktop"),
+              ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "games", "kde-desktop", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("KDE Desktop (Minimal)"),
+              ["base-x", "core", "kde-desktop-minimal",
+                "hardware-support"]),
+             (N_("XFCE Desktop"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "xfce-desktop", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("LXDE Desktop"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "lxde-desktop", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("Enlightenment Windows Manager"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "window-manager", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
              (N_("Software Development"),
               ["base", "base-x", "core", "development-libs",
                "development-tools", "editors", "fonts", "gnome-desktop",
@@ -100,7 +126,9 @@
 
         productUpgrades = {
                 "Fedora Core": ("Red Hat Linux", ),
-                "Fedora": ("Fedora Core", )
+                "Fedora": ("Fedora Core", ),
+                "RFRemix": ("Tedora", ),
+                "RFRemix": ("Fedora", )
         }
 
         if productUpgrades.has_key(productName):
@@ -116,8 +144,8 @@
 
     def versionMatches(self, oldver):
         try:
-            oldVer = float(oldver)
-            newVer = float(productVersion)
+            oldVer = float(oldver.split(".")[0])
+            newVer = float(productVersion.split(".")[0])
         except ValueError:
             return True
 
diff -urN anaconda-15.29.orig/pyanaconda/packages.py anaconda-15.29/pyanaconda/packages.py
--- anaconda-15.29.orig/pyanaconda/packages.py	2011-04-26 18:57:28.988824499 +0400
+++ anaconda-15.29/pyanaconda/packages.py	2011-04-26 18:59:53.225415502 +0400
@@ -303,7 +303,10 @@
     publicBetas = { "Red Hat Linux": "Red Hat Linux Public Beta",
                     "Red Hat Enterprise Linux": "Red Hat Enterprise Linux Public Beta",
                     "Fedora Core": "Fedora Core",
-                    "Fedora": "Fedora" }
+                    "Fedora": "Fedora",
+                    "Fedora": "RFRemix",
+                    "Tedora": "RFRemix",
+                    "RussianFedora": "RFRemix" }
 
     
     if anaconda.dir == DISPATCH_BACK:
