diff -urN anaconda-19.28.orig/data/liveinst/liveinst anaconda-19.28/data/liveinst/liveinst
--- anaconda-19.28.orig/data/liveinst/liveinst	2013-05-17 14:08:06.774513468 +0400
+++ anaconda-19.28/data/liveinst/liveinst	2013-05-17 14:08:44.849515971 +0400
@@ -19,7 +19,7 @@
 #
 
 if [ -n "$DISPLAY" -a -n "$LANG" ]; then
-    INSTLANG="--lang $LANG"
+    INSTLANG="--lang en_US.UTF-8"
 fi
 
 LIVE_INSTALL=0
