diff -urN anaconda-21.48.4.orig/pyanaconda/packaging/__init__.py anaconda-21.48.4/pyanaconda/packaging/__init__.py
--- anaconda-21.48.4.orig/pyanaconda/packaging/__init__.py	2014-09-08 17:38:18.245858020 +0400
+++ anaconda-21.48.4/pyanaconda/packaging/__init__.py	2014-09-08 17:39:45.612984098 +0400
@@ -677,7 +677,7 @@
 class PackagePayload(Payload):
     """ A PackagePayload installs a set of packages onto the target system. """
 
-    DEFAULT_REPOS = [productName.split('-')[0].lower(), "rawhide"]
+    DEFAULT_REPOS = ["fedora", "rawhide"]
 
     def __init__(self, data):
         if self.__class__ is PackagePayload:
