diff -urN anaconda-21.25.orig/data/liveinst/liveinst anaconda-21.25/data/liveinst/liveinst
--- anaconda-21.25.orig/data/liveinst/liveinst	2014-03-05 14:57:55.368934091 +0400
+++ anaconda-21.25/data/liveinst/liveinst	2014-03-05 15:01:17.460932206 +0400
@@ -66,15 +66,15 @@
 # load modules that would get loaded by the initramfs (#230945)
 for i in raid0 raid1 raid5 raid6 raid456 raid10 dm-mod dm-zero dm-mirror dm-snapshot dm-multipath dm-round-robin vfat dm-crypt cbc sha256 lrw xts iscsi_tcp iscsi_ibft; do /sbin/modprobe $i 2>/dev/null ; done
 
-export ANACONDA_PRODUCTNAME=$( cat /etc/system-release | sed -r -e 's/ *release.*//' )
+export ANACONDA_PRODUCTNAME=$( cat /etc/rfremix-release | sed -r -e 's/ *release.*//' )
 if [ $LIVE_INSTALL = 1 ]; then
-    export ANACONDA_PRODUCTVERSION=$( cat /etc/system-release | sed -r -e 's/^.* ([0-9\.]+).*$/\1/' )
+    export ANACONDA_PRODUCTVERSION=$( cat /etc/rfremix-release | sed -r -e 's/^.* ([0-9\.]+).*$/\1/' )
 elif [ $IMAGE_INSTALL = 1 ]; then
     export ANACONDA_PRODUCTVERSION=$(rpmquery -q --qf '%{VERSION}' anaconda | cut -d. -f1)
 fi
 export ANACONDA_BUGURL=${ANACONDA_BUGURL:="https://bugzilla.redhat.com/bugzilla/"}
 
-RELEASE=$(rpm -q --qf '%{Release}' fedora-release)
+RELEASE=$(rpm -q --qf '%{Release}' rfremix-release)
 if [ "${RELEASE:0:2}" = "0." ]; then
     export ANACONDA_ISFINAL="false"
 else
diff -urN anaconda-21.25.orig/pyanaconda/bootloader.py anaconda-21.25/pyanaconda/bootloader.py
--- anaconda-21.25.orig/pyanaconda/bootloader.py	2014-03-05 14:57:55.380934328 +0400
+++ anaconda-21.25/pyanaconda/bootloader.py	2014-03-05 15:01:51.589607816 +0400
@@ -1469,7 +1469,7 @@
         defaults_file = "%s%s" % (ROOT_PATH, self.defaults_file)
         defaults = open(defaults_file, "w+")
         defaults.write("GRUB_TIMEOUT=%d\n" % self.timeout)
-        defaults.write("GRUB_DISTRIBUTOR=\"$(sed 's, release .*$,,g' /etc/system-release)\"\n")
+        defaults.write("GRUB_DISTRIBUTOR=\"$(test -f /etc/rfremix-release && sed 's, release .*$,,g' /etc/rfremix-release || sed 's, release .*$,,g' /etc/system-release)\"\n")
         defaults.write("GRUB_DEFAULT=saved\n")
         defaults.write("GRUB_DISABLE_SUBMENU=true\n")
         if self.console and self.has_serial_console:
