diff -urN anaconda-21.44.orig/pyanaconda/packaging/__init__.py anaconda-21.44/pyanaconda/packaging/__init__.py
--- anaconda-21.44.orig/pyanaconda/packaging/__init__.py	2014-06-26 18:57:36.062036917 +0400
+++ anaconda-21.44/pyanaconda/packaging/__init__.py	2014-06-26 19:02:03.773497242 +0400
@@ -678,7 +678,7 @@
 class PackagePayload(Payload):
     """ A PackagePayload installs a set of packages onto the target system. """
 
-    DEFAULT_REPOS = [productName.lower(), "rawhide"]
+    DEFAULT_REPOS = ["fedora", "rawhide"]
 
     def __init__(self, data):
         if self.__class__ is PackagePayload:
