diff -urN anaconda-14.20.orig/scripts/upd-instroot anaconda-14.20/scripts/upd-instroot
--- anaconda-14.20.orig/scripts/upd-instroot	2010-10-15 18:56:01.889792053 +0400
+++ anaconda-14.20/scripts/upd-instroot	2010-10-18 17:33:07.804855667 +0400
@@ -155,7 +155,9 @@
     exit 1
 }
 
-PACKAGES="GConf2 NetworkManager NetworkManager-gnome ORBit2 acl anaconda
+PACKAGES="GConf2 NetworkManager NetworkManager-gnome NetworkManager-pptp NetworkManager-openswan
+    NetworkManager-vpnc NetworkManager-openconnect NetworkManager-openvpn
+    ORBit2 acl anaconda
     anaconda-yum-plugins at-spi atk attr audit-libs bash bind-utils
     bitmap-fonts-cjk
     btrfs-progs bzip2 bzip2-libs ca-certificates cairo cjkuni-uming-fonts
@@ -207,7 +209,11 @@
     xorg-x11-drivers xorg-x11-font-utils xorg-x11-fonts-ethiopic
     xorg-x11-fonts-misc xorg-x11-server-utils
     xorg-x11-server-Xorg xorg-x11-xkb-utils xorg-x11-xfs yum
-    yum-metadata-parser zenity zlib /etc/gtk-2.0/gtkrc"
+    yum-metadata-parser zenity zlib /etc/gtk-2.0/gtkrc
+    ar9170-firmware atmel-firmware b43-firmware ipw2100-firmware ipw2200-firmware
+    iwl1000-firmware iwl3945-firmware iwl4965-firmware iwl5000-firmware
+    iwl5150-firmware iwl6000-firmware iwl6050-firmware libertas-sd8686-firmware
+    libertas-usb8388-firmware rt61pci-firmware rt73usb-firmware zd1211-firmware"
 
 if [ $ARCH = i386 ]; then
     PACKAGES="$PACKAGES glibc.i386 openssl.i386"
