diff -up anaconda-11.5.0.23/scripts/upd-instroot.tigervnc anaconda-11.5.0.23/scripts/upd-instroot
--- anaconda-11.5.0.23/scripts/upd-instroot.tigervnc	2009-03-03 21:47:58.000000000 +0100
+++ anaconda-11.5.0.23/scripts/upd-instroot	2009-03-03 21:48:18.000000000 +0100
@@ -183,8 +183,8 @@ PACKAGES="GConf2 NetworkManager ORBit2 P
     reiserfs-utils rhpl rpm rpm-libs rpm-python sed selinux-policy-targeted
     setup slang smc-fonts-meera specspo sqlite synaptics system-config-date
     system-config-keyboard ${brandpkgname}-logos ${brandpkgname}-release taipeifonts tcp_wrappers
-    tzdata udev un-core-fonts-dotum urw-fonts util-linux-ng tightvnc-server
-    tightvnc-server-module vlgothic-fonts
+    tzdata udev un-core-fonts-dotum urw-fonts util-linux-ng tigervnc-server
+    tigervnc-server-module vlgothic-fonts
     wpa_supplicant xkeyboard-config xfsdump xfsprogs xorg-x11 xorg-x11-auth
     xorg-x11-base xorg-x11-drivers xorg-x11-font-utils xorg-x11-fonts-ethiopic
     xorg-x11-fonts-misc xorg-x11-libs xorg-x11-libs-data xorg-x11-server-utils
