diff -urN anaconda-14.17.4.orig/data/bootdisk/i386/syslinux.cfg anaconda-14.17.4/data/bootdisk/i386/syslinux.cfg
--- anaconda-14.17.4.orig/data/bootdisk/i386/syslinux.cfg	2010-09-28 21:39:32.007010938 +0400
+++ anaconda-14.17.4/data/bootdisk/i386/syslinux.cfg	2010-10-13 15:21:23.377775213 +0400
@@ -19,15 +19,15 @@
   menu label ^Install a new system or upgrade an existing system
   menu default
   kernel vmlinuz
-  append initrd=initrd.img
+  append initrd=initrd.img pcie_aspm=off
 label vesa
   menu label Install system with ^basic video driver
   kernel vmlinuz
-  append initrd=initrd.img xdriver=vesa nomodeset
+  append initrd=initrd.img xdriver=vesa nomodeset pcie_aspm=off
 label rescue
   menu label ^Rescue installed system
   kernel vmlinuz
-  append initrd=initrd.img rescue
+  append initrd=initrd.img rescue pcie_aspm=off
 label local
   menu label Boot from ^local drive
   localboot 0xffff
diff -urN anaconda-14.17.4.orig/data/bootdisk/x86_64/syslinux.cfg anaconda-14.17.4/data/bootdisk/x86_64/syslinux.cfg
--- anaconda-14.17.4.orig/data/bootdisk/x86_64/syslinux.cfg	2010-09-28 21:39:32.007010938 +0400
+++ anaconda-14.17.4/data/bootdisk/x86_64/syslinux.cfg	2010-10-13 15:21:38.882775850 +0400
@@ -19,15 +19,15 @@
   menu label ^Install a new system or upgrade an existing system
   menu default
   kernel vmlinuz
-  append initrd=initrd.img
+  append initrd=initrd.img pcie_aspm=off
 label vesa
   menu label Install system with ^basic video driver
   kernel vmlinuz
-  append initrd=initrd.img xdriver=vesa nomodeset
+  append initrd=initrd.img xdriver=vesa nomodeset pcie_aspm=off
 label rescue
   menu label ^Rescue installed system
   kernel vmlinuz
-  append initrd=initrd.img rescue
+  append initrd=initrd.img rescue pcie_aspm=off
 label local
   menu label Boot from ^local drive
   localboot 0xffff
diff -urN anaconda-14.17.4.orig/pyanaconda/packages.py anaconda-14.17.4/pyanaconda/packages.py
--- anaconda-14.17.4.orig/pyanaconda/packages.py	2010-09-28 21:39:31.983073381 +0400
+++ anaconda-14.17.4/pyanaconda/packages.py	2010-10-13 15:17:57.199906559 +0400
@@ -278,7 +278,7 @@
     if iutil.isConsoleOnVirtualTerminal() and \
        (ts.dbMatch('provides', 'rhgb').count() or \
         ts.dbMatch('provides', 'plymouth').count()):
-        anaconda.bootloader.args.append("rhgb quiet")
+        anaconda.bootloader.args.append("rhgb quiet pcie_aspm=off")
 
     if ts.dbMatch('provides', 'service(graphical-login)').count() and \
        ts.dbMatch('provides', 'xorg-x11-server-Xorg').count() and \
