diff -urN anaconda-15.31.orig/data/liveinst/liveinst anaconda-15.31/data/liveinst/liveinst
--- anaconda-15.31.orig/data/liveinst/liveinst	2011-05-20 13:55:29.669175329 +0400
+++ anaconda-15.31/data/liveinst/liveinst	2011-05-20 14:10:43.148799217 +0400
@@ -71,7 +71,7 @@
 fi
 export ANACONDA_BUGURL=${ANACONDA_BUGURL:="https://bugzilla.redhat.com/bugzilla/"}
 
-RELEASE=$(rpm -q --qf '%{Release}' fedora-release)
+RELEASE=$(rpm -q --qf '%{Release}' rfremix-release)
 if [ "${RELEASE:0:2}" = "0." ]; then
     export ANACONDA_ISBETA="true"
 else
