diff -urN anaconda-21.48.7.orig/pyanaconda/packaging/__init__.py anaconda-21.48.7/pyanaconda/packaging/__init__.py
--- anaconda-21.48.7.orig/pyanaconda/packaging/__init__.py	2014-09-19 15:40:34.349396393 +0400
+++ anaconda-21.48.7/pyanaconda/packaging/__init__.py	2014-09-19 15:41:18.121328437 +0400
@@ -684,7 +684,7 @@
 class PackagePayload(Payload):
     """ A PackagePayload installs a set of packages onto the target system. """
 
-    DEFAULT_REPOS = [productName.lower(), "rawhide"]
+    DEFAULT_REPOS = ["fedora", "rawhide"]
 
     def __init__(self, data):
         if self.__class__ is PackagePayload:
