diff -urN anaconda-13.21.82.orig/scripts/upd-instroot anaconda-13.21.82/scripts/upd-instroot
--- anaconda-13.21.82.orig/scripts/upd-instroot	2011-03-26 23:35:41.737798879 +0300
+++ anaconda-13.21.82/scripts/upd-instroot	2011-03-26 23:48:31.495305592 +0300
@@ -203,7 +203,13 @@
     xorg-x11-drivers xorg-x11-font-utils xorg-x11-fonts-ethiopic
     xorg-x11-fonts-misc xorg-x11-server-utils
     xorg-x11-server-Xorg xorg-x11-xkb-utils xorg-x11-xfs yum
-    yum-metadata-parser zenity zlib /etc/gtk-2.0/gtkrc"
+    yum-metadata-parser zenity zlib /etc/gtk-2.0/gtkrc
+    atmel-firmware b43-firmware ipw2100-firmware ipw2200-firmware
+    iwl1000-firmware iwl3945-firmware iwl4965-firmware iwl5000-firmware
+    iwl5150-firmware iwl6000-firmware iwl6050-firmware
+    libertas-usb8388-firmware rt61pci-firmware rt73usb-firmware zd1211-firmware
+    NetworkManager-openswan NetworkManager-openvpn NetworkManager-pptp
+    NetworkManager-vpnc"
 
 if [[ $ARCH =~ "i[356]86" ]]; then
     PACKAGES="$PACKAGES glibc.$ARCH openssl.$ARCH"
