diff -urN anaconda-25.20.3.orig/data/liveinst/gnome/fedora-welcome.desktop anaconda-25.20.3/data/liveinst/gnome/fedora-welcome.desktop
--- anaconda-25.20.3.orig/data/liveinst/gnome/fedora-welcome.desktop	2016-09-28 13:29:36.361917646 +0300
+++ anaconda-25.20.3/data/liveinst/gnome/fedora-welcome.desktop	2016-09-28 13:36:19.520019378 +0300
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Name=Welcome to Fedora
+Name=Welcome to RFRemix
 Exec=/usr/share/anaconda/gnome/fedora-welcome
 Terminal=false
 Type=Application
diff -urN anaconda-25.20.3.orig/data/liveinst/gnome/fedora-welcome.js anaconda-25.20.3/data/liveinst/gnome/fedora-welcome.js
--- anaconda-25.20.3.orig/data/liveinst/gnome/fedora-welcome.js	2016-09-28 13:29:36.361917646 +0300
+++ anaconda-25.20.3/data/liveinst/gnome/fedora-welcome.js	2016-09-28 13:36:00.000877357 +0300
@@ -59,7 +59,7 @@
                                                 default_width: 600,
                                                 default_height: 550,
                                                 skip_taskbar_hint: true,
-                                                title: _("Welcome to Fedora"),
+                                                title: _("Welcome to RFRemix"),
                                                 window_position: Gtk.WindowPosition.CENTER });
       this.window.connect('key-press-event', Lang.bind(this,
           function(w, event) {
@@ -88,7 +88,7 @@
                                      spacing: 16 });
       tryContent.add(new Gtk.Image({ icon_name: 'media-optical',
                                      pixel_size: 256 }));
-      tryContent.add(makeLabel(_("Try Fedora"), true));
+      tryContent.add(makeLabel(_("Try RFRemix"), true));
 
       let tryButton = new Gtk.Button({ child: tryContent });
       buttonBox.add(tryButton);
@@ -104,7 +104,7 @@
       let installButton = new Gtk.Button({ child: installContent });
       buttonBox.add(installButton);
 
-      this._label = makeLabel(_("You are currently running Fedora from live media.\nYou can install Fedora now, or choose \"Install to Hard Drive\" in the Activities Overview at any later time."), false);
+      this._label = makeLabel(_("You are currently running RFRemix from live media.\nYou can install RFRemix now, or choose \"Install to Hard Drive\" in the Activities Overview at any later time."), false);
       mainGrid.add(this._label);
 
       installButton.connect('clicked', Lang.bind(this,
