Только в anaconda-26.21.5: anaconda-26.21.5-hardcode-repo.patch
diff -ur anaconda-26.21.5.orig/pyanaconda/payload/__init__.py anaconda-26.21.5/pyanaconda/payload/__init__.py
--- anaconda-26.21.5.orig/pyanaconda/payload/__init__.py	2017-05-05 16:22:01.825089486 +0300
+++ anaconda-26.21.5/pyanaconda/payload/__init__.py	2017-05-05 16:23:54.009522400 +0300
@@ -787,7 +787,7 @@
 class PackagePayload(Payload):
     """A PackagePayload installs a set of packages onto the target system."""
 
-    DEFAULT_REPOS = [productName.split('-')[0].lower(), "rawhide"]          # pylint: disable=no-member
+    DEFAULT_REPOS = ["fedora", "rawhide"]          # pylint: disable=no-member
 
     def __init__(self, data):
         if self.__class__ is PackagePayload:
