diff -urN anaconda-20.25.6.orig/pyanaconda/packaging/dnfpayload.py anaconda-20.25.6/pyanaconda/packaging/dnfpayload.py
--- anaconda-20.25.6.orig/pyanaconda/packaging/dnfpayload.py	2013-11-07 13:36:32.563839128 +0400
+++ anaconda-20.25.6/pyanaconda/packaging/dnfpayload.py	2013-11-07 13:43:20.667721864 +0400
@@ -46,7 +46,7 @@
     dnf = None
     rpm = None
 
-DEFAULT_REPOS = [constants.productName.lower(), "rawhide"]
+DEFAULT_REPOS = ["fedora", "rawhide"]
 DNF_CACHE_DIR = '/tmp/dnf.cache'
 REPO_DIRS = ['/etc/yum.repos.d',
              '/etc/anaconda.repos.d',
diff -urN anaconda-20.25.6.orig/pyanaconda/packaging/yumpayload.py anaconda-20.25.6/pyanaconda/packaging/yumpayload.py
--- anaconda-20.25.6.orig/pyanaconda/packaging/yumpayload.py	2013-11-07 13:36:32.564839152 +0400
+++ anaconda-20.25.6/pyanaconda/packaging/yumpayload.py	2013-11-07 13:41:31.149083762 +0400
@@ -87,7 +87,7 @@
 from pykickstart.constants import GROUP_ALL, GROUP_DEFAULT, KS_MISSING_IGNORE
 
 YUM_PLUGINS = ["blacklist", "whiteout", "fastestmirror", "langpacks"]
-DEFAULT_REPOS = [productName.lower(), "rawhide"]
+DEFAULT_REPOS = ["fedora", "rawhide"]
 BASE_REPO_NAMES = [BASE_REPO_NAME] + DEFAULT_REPOS
 
 import inspect
@@ -162,7 +162,7 @@
 
         PackagePayload.__init__(self, data)
 
-        self.default_repos = [productName.lower(), "rawhide"]
+        self.default_repos = ["fedora", "rawhide"]
 
         self._root_dir = "/tmp/yum.root"
         self._repos_dir = "/etc/yum.repos.d,/etc/anaconda.repos.d,/tmp/updates/anaconda.repos.d,/tmp/product/anaconda.repos.d"
