diff -urN anaconda-16.22.orig/po/ru.po anaconda-16.22/po/ru.po
--- anaconda-16.22.orig/po/ru.po	2011-10-21 13:09:21.548137862 +0400
+++ anaconda-16.22/po/ru.po	2011-10-21 13:14:45.211495636 +0400
@@ -2205,18 +2205,42 @@
 "хотели бы включить?"
 
 #: pyanaconda/installclasses/fedora.py:50
-msgid "Graphical Desktop"
-msgstr "Графический рабочий стол"
+msgid "GNOME Desktop"
+msgstr "Рабочий стол GNOME"
 
 #: pyanaconda/installclasses/fedora.py:55
+msgid "GNOME Desktop (Minimal)"
+msgstr "Рабочий стол GNOME (минимально)"
+
+#: pyanaconda/installclasses/fedora.py:58
+msgid "KDE Desktop"
+msgstr "Рабочий стол KDE"
+
+#: pyanaconda/installclasses/fedora.py:63
+msgid "KDE Desktop (Minimal)"
+msgstr "Рабочий стол KDE (минимально)"
+
+#: pyanaconda/installclasses/fedora.py:66
+msgid "XFCE Desktop"
+msgstr "Рабочий стол XFCE"
+
+#: pyanaconda/installclasses/fedora.py:71
+msgid "LXDE Desktop"
+msgstr "Рабочий стол LXDE"
+
+#: pyanaconda/installclasses/fedora.py:76
+msgid "Enlightenment Window Manager"
+msgstr "Оконный менеджер Enlightenment"
+
+#: pyanaconda/installclasses/fedora.py:81
 msgid "Software Development"
 msgstr "Разработка ПО"
 
-#: pyanaconda/installclasses/fedora.py:61
+#: pyanaconda/installclasses/fedora.py:81
 msgid "Web Server"
 msgstr "Веб-сервер"
 
-#: pyanaconda/installclasses/fedora.py:65 pyanaconda/installclasses/rhel.py:44
+#: pyanaconda/installclasses/fedora.py:91 pyanaconda/installclasses/rhel.py:44
 msgid "Minimal"
 msgstr "Минимальный"
 
diff -urN anaconda-16.22.orig/pyanaconda/installclasses/fedora.py anaconda-16.22/pyanaconda/installclasses/fedora.py
--- anaconda-16.22.orig/pyanaconda/installclasses/fedora.py	2011-10-21 13:09:21.561138102 +0400
+++ anaconda-16.22/pyanaconda/installclasses/fedora.py	2011-10-21 13:14:38.528371366 +0400
@@ -36,8 +36,8 @@
 
 class InstallClass(BaseInstallClass):
     # name has underscore used for mnemonics, strip if you dont need it
-    id = "fedora"
-    name = N_("_Fedora")
+    id = "rfremix"
+    name = N_("_RFRemix")
     _description = N_("The default installation of %s includes a set of "
                       "software applicable for general internet usage. "
                       "You can optionally select a different set of software "
@@ -47,11 +47,37 @@
     if productName.startswith("Red Hat Enterprise"):
         hidden = 1
 
-    tasks = [(N_("Graphical Desktop"),
+    tasks = [(N_("GNOME Desktop"),
               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
                "games", "gnome-desktop", "graphical-internet", "graphics",
                "hardware-support", "input-methods", "java", "office",
                "printing", "sound-and-video", "text-internet"]),
+             (N_("GNOME Desktop (Minimal)"),
+              ["base-x", "core", "gnome-desktop-minimal",
+               "hardware-support"]),
+             (N_("KDE Desktop"),
+              ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "games", "kde-desktop", "desktop-minimal",
+               "hardware-support", "input-methods", "java",
+               "printing", "text-internet"]),
+             (N_("KDE Desktop (Minimal)"),
+              ["base-x", "core", "kde-desktop-minimal",
+                "hardware-support"]),
+             (N_("XFCE Desktop"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "xfce-desktop", "desktop-minimal", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("LXDE Desktop"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "lxde-desktop", "desktop-minimal", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("Enlightenment Window Manager"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "window-manager", "desktop-minimal", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
              (N_("Software Development"),
               ["base", "base-x", "core", "development-libs",
                "development-tools", "editors", "fonts", "gnome-desktop",
@@ -98,7 +124,9 @@
 
         productUpgrades = {
                 "Fedora Core": ("Red Hat Linux", ),
-                "Fedora": ("Fedora Core", )
+                "Fedora": ("Fedora Core", ),
+                "RFRemix": ("Tedora", ),
+                "RFRemix": ("Fedora", )
         }
 
         if productUpgrades.has_key(productName):
@@ -114,9 +142,9 @@
 
     def versionMatches(self, oldver):
         try:
-            oldVer = float(oldver)
+            oldVer = float(oldver.split(".")[0])
             # Trim off any "-Alpha" or "-Beta".
-            newVer = float(productVersion.split('-')[0])
+            newVer = float(productVersion.split('-')[0].split(".")[0])
         except ValueError:
             return True
 
diff -urN anaconda-16.22.orig/pyanaconda/packages.py anaconda-16.22/pyanaconda/packages.py
--- anaconda-16.22.orig/pyanaconda/packages.py	2011-10-21 13:09:21.568138232 +0400
+++ anaconda-16.22/pyanaconda/packages.py	2011-10-21 13:09:33.237354357 +0400
@@ -308,7 +308,11 @@
     publicBetas = { "Red Hat Linux": "Red Hat Linux Public Beta",
                     "Red Hat Enterprise Linux": "Red Hat Enterprise Linux Public Beta",
                     "Fedora Core": "Fedora Core",
-                    "Fedora": "Fedora" }
+                    "Fedora": "Fedora",
+                    "Fedora": "Fedora",
+                    "Fedora": "RFRemix",
+                    "Tedora": "RFRemix",
+                    "RussianFedora": "RFRemix" }
 
 
     if anaconda.dir == DISPATCH_BACK:
