diff -urN anaconda-13.21.82.orig/bootdisk/i386/syslinux.cfg anaconda-13.21.82/bootdisk/i386/syslinux.cfg
--- anaconda-13.21.82.orig/bootdisk/i386/syslinux.cfg	2011-03-28 19:03:29.080059816 +0400
+++ anaconda-13.21.82/bootdisk/i386/syslinux.cfg	2011-03-28 19:26:18.156907393 +0400
@@ -15,9 +15,13 @@
 menu color hotkey 7 #ffffffff #ff000000
 menu color scrollbar 0 #ffffffff #00000000
 
+label linux-quick
+  menu label ^Quick install
+  menu default
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/install.cfg
 label linux
   menu label ^Install or upgrade an existing system
-  menu default
   kernel vmlinuz
   append initrd=initrd.img
 label vesa
diff -urN anaconda-13.21.82.orig/bootdisk/x86_64/syslinux.cfg anaconda-13.21.82/bootdisk/x86_64/syslinux.cfg
--- anaconda-13.21.82.orig/bootdisk/x86_64/syslinux.cfg	2011-03-28 19:03:29.081059715 +0400
+++ anaconda-13.21.82/bootdisk/x86_64/syslinux.cfg	2011-03-28 19:26:18.156907393 +0400
@@ -15,9 +15,13 @@
 menu color hotkey 7 #ffffffff #ff000000
 menu color scrollbar 0 #ffffffff #00000000
 
+label linux-quick
+  menu label ^Quick install
+  menu default
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/install.cfg
 label linux
   menu label ^Install or upgrade an existing system
-  menu default
   kernel vmlinuz
   append initrd=initrd.img
 label vesa
diff -urN anaconda-13.21.82.orig/scripts/mk-images anaconda-13.21.82/scripts/mk-images
--- anaconda-13.21.82.orig/scripts/mk-images	2011-03-28 19:03:29.114056385 +0400
+++ anaconda-13.21.82/scripts/mk-images	2011-03-28 19:42:43.882603953 +0400
@@ -887,6 +887,45 @@
 
     setupSshd
 
+    cat > $MBD_DIR/install.cfg <<EOF
+install
+lang ru_RU.UTF-8
+keyboard ru
+timezone Europe/Moscow
+firstboot --enable
+%packages
+@base
+@basic-desktop
+@cifs-file-server
+@desktop-platform
+@directory-client
+@fonts
+@general-desktop
+@graphical-admin-tools
+@graphics
+@input-methods
+@internet-applications
+@internet-browser
+@java-platform
+@legacy-x
+@mail-server
+@mysql
+@network-file-system-client
+@network-server
+@network-tools
+@office-suite
+@print-client
+@print-server
+@remote-desktop-clients
+@russian-support
+@server-platform
+@web-server
+@x11
+-kexec-tools
+-system-config-kdump
+%end
+EOF
+
 cat > $MBD_DIR/.profile <<EOF
 PS1="[anaconda \u@\h \W]\\\\$ "
 PATH=/bin:/usr/bin:/usr/sbin:/mnt/sysimage/sbin:/mnt/sysimage/usr/sbin:/mnt/sysimage/bin:/mnt/sysimage/usr/bin
