diff -urN anaconda-19.16.orig/pyanaconda/packaging/yumpayload.py anaconda-19.16/pyanaconda/packaging/yumpayload.py
--- anaconda-19.16.orig/pyanaconda/packaging/yumpayload.py	2013-04-09 12:11:35.863032084 +0400
+++ anaconda-19.16/pyanaconda/packaging/yumpayload.py	2013-04-09 12:19:54.863635676 +0400
@@ -93,7 +93,7 @@
 from pykickstart.constants import KS_MISSING_IGNORE
 
 YUM_PLUGINS = ["blacklist", "whiteout", "fastestmirror", "langpacks"]
-default_repos = [productName.lower(), "rawhide"]
+default_repos = ["fedora", "rawhide"]
 
 import inspect
 import threading
