diff -urN anaconda-26.21.1.orig/pyanaconda/payload/__init__.py anaconda-26.21.1/pyanaconda/payload/__init__.py
--- anaconda-26.21.1.orig/pyanaconda/payload/__init__.py	2017-03-09 18:29:49.932742782 +0300
+++ anaconda-26.21.1/pyanaconda/payload/__init__.py	2017-03-09 18:31:26.970638237 +0300
@@ -738,7 +738,7 @@
 class PackagePayload(Payload):
     """ A PackagePayload installs a set of packages onto the target system. """
 
-    DEFAULT_REPOS = [productName.split('-')[0].lower(), "rawhide"]          # pylint: disable=no-member
+    DEFAULT_REPOS = ["fedora", "rawhide"]          # pylint: disable=no-member
 
     def __init__(self, data):
         if self.__class__ is PackagePayload:
