diff -urN anaconda-18.8.orig/pyanaconda/installclasses/fedora.py anaconda-18.8/pyanaconda/installclasses/fedora.py
--- anaconda-18.8.orig/pyanaconda/installclasses/fedora.py	2012-09-24 14:34:43.539398284 +0400
+++ anaconda-18.8/pyanaconda/installclasses/fedora.py	2012-09-24 14:50:30.312157811 +0400
@@ -31,8 +31,8 @@
 
 class InstallClass(BaseInstallClass):
     # name has underscore used for mnemonics, strip if you dont need it
-    id = "fedora"
-    name = N_("_Fedora")
+    id = "rfremix"
+    name = N_("_RFRemix")
     _description = N_("The default installation of %s includes a set of "
                       "software applicable for general internet usage. "
                       "You can optionally select a different set of software "
@@ -86,7 +86,9 @@
 
         productUpgrades = {
                 "Fedora Core": ("Red Hat Linux", ),
-                "Fedora": ("Fedora Core", )
+                "Fedora": ("Fedora Core", ),
+                "RFRemix": ("Tedora", ),
+                "RFRemix": ("Fedora", )
         }
 
         if productUpgrades.has_key(productName):
@@ -105,9 +107,9 @@
             return False
 
         try:
-            oldVer = Decimal(oldver)
+            oldVer = Decimal(oldver.split(".")[0])
             # Trim off any "-Alpha" or "-Beta".
-            newVer = Decimal(productVersion.split('-')[0])
+            newVer = Decimal(productVersion.split('-')[0].split(".")[0])
         except Exception:
             return True
 
