diff -urN anaconda-14.17.4.orig/po/ru.po anaconda-14.17.4/po/ru.po
--- anaconda-14.17.4.orig/po/ru.po	2010-09-28 21:39:31.958073834 +0400
+++ anaconda-14.17.4/po/ru.po	2010-09-28 21:41:12.153026216 +0400
@@ -2261,19 +2261,39 @@
 "обычных задач работы с Интернетом. Поддержку каких дополнительных задач вы "
 "хотели бы включить?"
 
-#: pyanaconda/installclasses/fedora.py:48
-msgid "Graphical Desktop"
-msgstr "Графический рабочий стол"
+#: installclasses/fedora.py:48
+msgid "GNOME Desktop"
+msgstr "Рабочий стол GNOME"
+
+#: installclasses/fedora.py:53
+msgid "GNOME Desktop (Minimal)"
+msgstr "Рабочий стол GNOME (минимально)"
+
+#: installclasses/fedora.py:56
+msgid "KDE Desktop"
+msgstr "Рабочий стол KDE"
+
+#: installclasses/fedora.py:61
+msgid "KDE Desktop (Minimal)"
+msgstr "Рабочий стол KDE (минимально)"
+
+#: installclasses/fedora.py:64
+msgid "XFCE Desktop"
+msgstr "Рабочий стол XFCE"
+
+#: installclasses/fedora.py:69
+msgid "LXDE Desktop"
+msgstr "Рабочий стол LXDE"
 
-#: pyanaconda/installclasses/fedora.py:53
+#: pyanaconda/installclasses/fedora.py:74
 msgid "Software Development"
 msgstr "Разработка ПО"
 
-#: pyanaconda/installclasses/fedora.py:59
+#: pyanaconda/installclasses/fedora.py:80
 msgid "Web Server"
 msgstr "Веб-сервер"
 
-#: pyanaconda/installclasses/fedora.py:63 pyanaconda/installclasses/rhel.py:44
+#: pyanaconda/installclasses/fedora.py:84 pyanaconda/installclasses/rhel.py:44
 msgid "Minimal"
 msgstr "Минимальный"
 
diff -urN anaconda-14.17.4.orig/pyanaconda/installclasses/fedora.py anaconda-14.17.4/pyanaconda/installclasses/fedora.py
--- anaconda-14.17.4.orig/pyanaconda/installclasses/fedora.py	2010-09-28 21:39:31.974073302 +0400
+++ anaconda-14.17.4/pyanaconda/installclasses/fedora.py	2010-09-28 21:42:07.809012405 +0400
@@ -34,8 +34,8 @@
 
 class InstallClass(BaseInstallClass):
     # name has underscore used for mnemonics, strip if you dont need it
-    id = "fedora"
-    name = N_("_Fedora")
+    id = "russianfedora"
+    name = N_("_RFRemix")
     _description = N_("The default installation of %s includes a set of "
                       "software applicable for general internet usage. "
                       "You can optionally select a different set of software "
@@ -45,11 +45,32 @@
     if productName.startswith("Red Hat Enterprise"):
         hidden = 1
 
-    tasks = [(N_("Graphical Desktop"),
+    tasks = [(N_("GNOME Desktop"),
               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
                "games", "gnome-desktop", "graphical-internet", "graphics",
                "hardware-support", "input-methods", "java", "office",
                "printing", "sound-and-video", "text-internet"]),
+             (N_("GNOME Desktop (Minimal)"),
+              ["admin-tools", "base", "base-x", "core", "gnome-desktop-minimal",
+               "hardware-support", "graphical-internet"]),
+             (N_("KDE Desktop"),
+              ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "games", "kde-desktop", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("KDE Desktop (Minimal)"),
+              ["admin-tools", "base", "base-x", "core", "kde-desktop-minimal",
+                "hardware-support"]),
+             (N_("XFCE Desktop"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "games", "xfce-desktop", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
+             (N_("LXDE Desktop"),
+               ["admin-tools", "base", "base-x", "core", "editors", "fonts",
+               "games", "lxde-desktop", "graphical-internet", "graphics",
+               "hardware-support", "input-methods", "java", "office",
+               "printing", "sound-and-video", "text-internet"]),
              (N_("Software Development"),
               ["base", "base-x", "core", "development-libs",
                "development-tools", "editors", "fonts", "gnome-desktop",
@@ -98,7 +119,9 @@
 
         productUpgrades = {
                 "Fedora Core": ("Red Hat Linux", ),
-                "Fedora": ("Fedora Core", )
+                "Fedora": ("Fedora Core", ),
+                "RFRemix": ("Tedora", ),
+                "RFRemix": ("Fedora", )
         }
 
         if productUpgrades.has_key(productName):
@@ -114,8 +137,8 @@
 
     def versionMatches(self, oldver):
         try:
-            oldVer = float(oldver)
-            newVer = float(productVersion)
+            oldVer = float(oldver.split(".")[0])
+            newVer = float(productVersion.split(".")[0])
         except ValueError:
             return True
 
diff -urN anaconda-14.17.4.orig/pyanaconda/packages.py anaconda-14.17.4/pyanaconda/packages.py
--- anaconda-14.17.4.orig/pyanaconda/packages.py	2010-09-28 21:39:31.983073381 +0400
+++ anaconda-14.17.4/pyanaconda/packages.py	2010-09-28 21:40:25.833136982 +0400
@@ -297,7 +297,10 @@
     publicBetas = { "Red Hat Linux": "Red Hat Linux Public Beta",
                     "Red Hat Enterprise Linux": "Red Hat Enterprise Linux Public Beta",
                     "Fedora Core": "Fedora Core",
-                    "Fedora": "Fedora" }
+                    "Fedora": "Fedora",
+                    "Fedora": "RFRemix",
+                    "Tedora": "RFRemix",
+                    "RussianFedora": "RFRemix" }
 
     
     if anaconda.dir == DISPATCH_BACK:
