diff -urN anaconda-13.21.117.orig/scripts/upd-instroot anaconda-13.21.117/scripts/upd-instroot
--- anaconda-13.21.117.orig/scripts/upd-instroot	2011-10-18 13:40:59.147985551 +0400
+++ anaconda-13.21.117/scripts/upd-instroot	2011-10-25 18:27:28.590193833 +0400
@@ -1134,6 +1134,69 @@
 mv $DEST/usr/lib/anaconda-runtime/lib* $DEST/usr/$LIBDIR 2>/dev/null
 mv $DEST/usr/lib/anaconda/sitecustomize.py $DEST/usr/$LIBDIR/python?.?/site-packages
 
+rm -f $DEST/etc/yum.repos.d/*
+cat > $DEST/etc/yum.repos.d/reremix-release-anaconda.repo <<EOF
+[reremix]
+name=RERemix \$releasever - \$basearch
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/build/6.1/reremix/\$basearch
+enabled=1
+gpgcheck=0
+
+[reremix-security]
+name=RERemix \$releasever - \$basearch - Updates
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/build/6.1/reremix-security/\$basearch
+enabled=1
+gpgcheck=0
+
+[dag]
+name=DAG for EL \$releasever - \$basearch
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/build/6.1/dag/\$basearch
+enabled=1
+gpgcheck=0
+
+[epel]
+name=EPEL for EL \$releasever - \$basearch
+baseurl=http://mirror.yandex.ru/epel/6/\$basearch
+enabled=1
+gpgcheck=0
+
+[russianfedora-fixes]
+name=Russian Fedora for EL \$releasever - Fixes
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/russianfedora/fixes/el/releases/6/Everything/\$basearch/os/
+enabled=1
+gpgcheck=0
+
+[russianfedora-fixes-updates]
+name=Russian Fedora for EL \$releasever - Fixes - Updates
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/russianfedora/fixes/el/updates/6/\$basearch
+enabled=1
+gpgcheck=0
+
+[russianfedora-free]
+name=Russian Fedora for EL \$releasever - Free
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/russianfedora/free/el/releases/6/Everything/\$basearch/os/
+enabled=1
+gpgcheck=0
+
+[russianfedora-free-updates]
+name=Russian Fedora for EL \$releasever - Free - Updates
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/russianfedora/free/el/updates/6/\$basearch
+enabled=1
+gpgcheck=0
+
+[russianfedora-nonfree]
+name=Russian Fedora for EL \$releasever - Non-Free 
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/russianfedora/nonfree/el/releases/6/Everything/\$basearch/os/
+enabled=1
+gpgcheck=0
+
+[russianfedora-nonfree-updates]
+name=Russian Fedora for EL \$releasever - Non-Free - Updates
+baseurl=http://mirror.yandex.ru/fedora/russianfedora/russianfedora/nonfree/el/updates/6/\$basearch
+enabled=1
+gpgcheck=0
+EOF
+
 mv $DEST/etc/yum.repos.d $DEST/etc/anaconda.repos.d
 
 rm -f $DEST/usr/$LIBDIR/libunicode-lite*
