diff -urN anaconda-15.22.orig/scripts/upd-instroot anaconda-15.22/scripts/upd-instroot
--- anaconda-15.22.orig/scripts/upd-instroot	2011-03-18 18:34:03.786218709 +0300
+++ anaconda-15.22/scripts/upd-instroot	2011-03-18 18:36:47.160603865 +0300
@@ -1259,6 +1259,149 @@
 
 setupSshd
 
+cat > $MBD_DIR/gnome.cfg <<EOF
+install
+lang ru_RU.UTF-8
+keyboard ru
+%packages
+@admin-tools
+@base
+@base-x
+@core
+@editors
+@fonts
+@gnome-desktop
+@graphical-internet
+@graphics
+@hardware-support
+@java
+@office
+@printing
+@russian-support
+@sound-and-video
+@system-tools
+@text-internet
+%end
+EOF
+
+cat > $MBD_DIR/kde.cfg <<EOF
+install
+lang ru_RU.UTF-8
+keyboard ru
+%packages
+@admin-tools
+@base
+@base-x
+@core
+@editors
+@fonts
+@games
+@kde-desktop
+@graphical-internet
+@graphics
+@hardware-support
+@input-methods
+@java
+@office
+@printing
+@russian-support
+@sound-and-video
+@text-internet
+%end
+EOF
+
+cat > $MBD_DIR/xfce.cfg <<EOF
+install
+lang ru_RU.UTF-8
+keyboard ru
+%packages
+@admin-tools
+@base
+@base-x
+@core
+@editors
+@fonts
+@games
+@xfce-desktop
+@graphical-internet
+@graphics
+@hardware-support
+@input-methods
+@java
+@office
+@printing
+@russian-support
+@sound-and-video
+@text-internet
+%end
+EOF
+
+cat > $MBD_DIR/lxde.cfg <<EOF
+install
+lang ru_RU.UTF-8
+keyboard ru
+%packages
+@admin-tools
+@base
+@base-x
+@core
+@editors
+@fonts
+@games
+@lxde-desktop
+@graphical-internet
+@graphics
+@hardware-support
+@input-methods
+@java
+@office
+@printing
+@russian-support
+@sound-and-video
+@text-internet
+%end
+EOF
+
+cat > $MBD_DIR/tigro.cfg <<EOF
+install
+lang ru_RU.UTF-8
+keyboard ru
+%packages
+@admin-tools
+@base
+@base-x
+@core
+@editors
+@fonts
+@gnome-desktop
+@graphical-internet
+@graphics
+@hardware-support
+@java
+@office
+@printing
+@russian-support
+@sound-and-video
+@system-tools
+@text-internet
+liferea
+gajim
+gwibber
+mail-notification
+ncftp
+goldendict
+k3b-extras-freeworld
+shutter
+lynx
+%end
+
+%post
+if [ -f /etc/sysconfig/keyboard ]; then
+sed -i 's!grp:alt_shift_toggle,grp_led:scroll!grp:rctrl_rshift_toggle,grp_led:scroll!' /etc/sysconfig/keyboard
+fi
+%end
+EOF
+
 sed -i 's|\(installforallkernels\) = 0|\1 = 1|' $DEST/etc/yum/pluginconf.d/fedorakmod.conf
 
 #
