diff -up anaconda-11.4.0.82/iutil.py.BAD anaconda-11.4.0.82/iutil.py
--- anaconda-11.4.0.82/iutil.py.BAD	2008-10-01 17:29:30.000000000 -0400
+++ anaconda-11.4.0.82/iutil.py	2008-10-01 17:29:34.000000000 -0400
@@ -538,7 +538,7 @@ def inXen():
 ## Check to see if we are in a vmware environment.
 #
 def inVmware():
-    out = execWithCapture("/usr/sbin/lspci", ["-vvv"])
+    out = execWithCapture("lspci", ["-vvv"])
     if "VMware" in out:
         return True
     return False
