diff -urN anaconda-13.21.82.orig/scripts/upd-instroot anaconda-13.21.82/scripts/upd-instroot
--- anaconda-13.21.82.orig/scripts/upd-instroot	2011-03-26 23:35:41.737798879 +0300
+++ anaconda-13.21.82/scripts/upd-instroot	2011-03-26 23:40:12.850419088 +0300
@@ -192,7 +192,7 @@
     python-libs python-nss python-pyblock python-sqlite python-epdb
     python-urlgrabber python-volume_key pyxf86config readline redhat-artwork
     reiserfs-utils report-config-default report-gtk report-newt rpm rpm-libs
-    rpm-python sed selinux-policy-targeted
+    rpm-python sed selinux-policy-mls
     setup slang smc-meera-fonts specspo sqlite startup-notification 
     synaptics system-config-date
     system-config-keyboard ${brandpkgname}-logos ${brandpkgname}-release
@@ -401,7 +401,7 @@
 etc/rpm/macros.prelink
 etc/security/limits.conf
 etc/security/pam_env.conf
-etc/selinux/targeted
+etc/selinux/mls
 etc/services
 etc/shells
 etc/sysconfig/network-scripts/network-functions*
@@ -1057,10 +1057,10 @@
 done
 
 # create selinux config
-if [ -e $DEST/etc/selinux/targeted ]; then
+if [ -e $DEST/etc/selinux/mls ]; then
   cat > $DEST/etc/selinux/config <<EOF
 SELINUX=permissive
-SELINUXTYPE=targeted
+SELINUXTYPE=mls
 EOF
 fi
 
diff -urN anaconda-13.21.82.orig/yuminstall.py anaconda-13.21.82/yuminstall.py
--- anaconda-13.21.82.orig/yuminstall.py	2011-03-26 23:35:41.863805672 +0300
+++ anaconda-13.21.82/yuminstall.py	2011-03-26 23:38:59.869482708 +0300
@@ -370,7 +370,7 @@
 
         if flags.selinux:
             for directory in ("/tmp/updates",
-                        "/etc/selinux/targeted/contexts/files",
+                        "/etc/selinux/mls/contexts/files",
                         "/etc/security/selinux/src/policy/file_contexts",
                         "/etc/security/selinux"):
                 fn = "%s/file_contexts" %(directory,)
