diff -urN anaconda-16.18.orig/data/bootdisk/i386/syslinux.cfg anaconda-16.18/data/bootdisk/i386/syslinux.cfg
--- anaconda-16.18.orig/data/bootdisk/i386/syslinux.cfg	2011-09-16 16:41:46.723430078 +0400
+++ anaconda-16.18/data/bootdisk/i386/syslinux.cfg	2011-09-16 16:52:20.033781903 +0400
@@ -24,6 +24,22 @@
   menu label Install system with ^basic video driver
   kernel vmlinuz
   append initrd=initrd.img xdriver=vesa nomodeset quiet
+label quick-gnome
+  menu label Quick install GNOME (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/gnome.cfg
+label quick-kde
+  menu label Quick install KDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/kde.cfg
+label quick-xfce
+  menu label Quick install XFCE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/xfce.cfg
+label quick-lxde
+  menu label Quick install LXDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/lxde.cfg
 label rescue
   menu label ^Rescue installed system
   kernel vmlinuz
diff -urN anaconda-16.18.orig/data/bootdisk/x86_64/syslinux.cfg anaconda-16.18/data/bootdisk/x86_64/syslinux.cfg
--- anaconda-16.18.orig/data/bootdisk/x86_64/syslinux.cfg	2011-09-16 16:41:46.725430113 +0400
+++ anaconda-16.18/data/bootdisk/x86_64/syslinux.cfg	2011-09-16 16:52:42.594186818 +0400
@@ -24,6 +24,22 @@
   menu label Install system with ^basic video driver
   kernel vmlinuz
   append initrd=initrd.img xdriver=vesa nomodeset quiet
+label quick-gnome
+  menu label Quick install GNOME (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/gnome.cfg
+label quick-kde
+  menu label Quick install KDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/kde.cfg
+label quick-xfce
+  menu label Quick install XFCE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/xfce.cfg
+label quick-lxde
+  menu label Quick install LXDE (in Russian)
+  kernel vmlinuz
+  append initrd=initrd.img ks=file:/lxde.cfg
 label rescue
   menu label ^Rescue installed system
   kernel vmlinuz
