From da565b769979a031f318dbc727b9888e4f1fb37c Mon Sep 17 00:00:00 2001
From: Chris Lumens <clumens@redhat.com>
Date: Mon, 06 May 2013 21:18:30 +0000
Subject: Revert "Add signal handlers for controlling password entry visibility." (#958608).

This reverts commit 99464761dab4e43cfbf8caa059815c6ab67c6282.
---
diff --git a/pyanaconda/ui/gui/__init__.py b/pyanaconda/ui/gui/__init__.py
index b0e04ef..bfe7877 100644
--- a/pyanaconda/ui/gui/__init__.py
+++ b/pyanaconda/ui/gui/__init__.py
@@ -198,24 +198,6 @@ class GUIObject(common.UIObject):
         """
         self.window.set_warning(msg)
 
-    def on_password_focused(self, entry, event, *args):
-        """When a password entry is focused, this callback allows for showing
-           the contents of the entry.  Along with on_password_unfocused, this
-           has the effect of hiding the entry when the user's not directly
-           editing it, and showing it when they are.
-        """
-        entry.set_visibility(True)
-        return False
-
-    def on_password_unfocused(self, entry, event, *args):
-        """When a password entry is unfocused, this callback allows for hiding
-           the contents of the entry.  Along with on_password_focused, this
-           has the effect of hiding the entry when the user's not directly
-           editing it, and showing it when they are.
-        """
-        entry.set_visibility(False)
-        return False
-
 class QuitDialog(GUIObject):
     builderObjects = ["quitDialog"]
     mainWidgetName = "quitDialog"
diff --git a/pyanaconda/ui/gui/spokes/custom.glade b/pyanaconda/ui/gui/spokes/custom.glade
index 21e2daa..bc92834 100644
--- a/pyanaconda/ui/gui/spokes/custom.glade
+++ b/pyanaconda/ui/gui/spokes/custom.glade
@@ -1383,8 +1383,6 @@ until you click on the main menu's 'Begin Installation' button.</property>
                                     <property name="visibility">False</property>
                                     <property name="invisible_char">●</property>
                                     <accelerator key="p" signal="grab-focus" modifiers="GDK_MOD1_MASK"/>
-                                    <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                                    <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                                   </object>
                                   <packing>
                                     <property name="expand">False</property>
diff --git a/pyanaconda/ui/gui/spokes/lib/passphrase.glade b/pyanaconda/ui/gui/spokes/lib/passphrase.glade
index aee8e5a..34ee0fa 100644
--- a/pyanaconda/ui/gui/spokes/lib/passphrase.glade
+++ b/pyanaconda/ui/gui/spokes/lib/passphrase.glade
@@ -117,8 +117,6 @@
                 <property name="width_chars">32</property>
                 <signal name="changed" handler="on_passphrase_changed" swapped="no"/>
                 <signal name="focus-out-event" handler="on_passphrase_editing_done" swapped="no"/>
-                <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
-                <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
               </object>
               <packing>
                 <property name="left_attach">1</property>
@@ -152,8 +150,6 @@
                 <property name="width_chars">32</property>
                 <signal name="changed" handler="on_confirm_changed" swapped="no"/>
                 <signal name="focus-out-event" handler="on_confirm_editing_done" swapped="no"/>
-                <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
-                <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
               </object>
               <packing>
                 <property name="left_attach">1</property>
diff --git a/pyanaconda/ui/gui/spokes/lib/passphrase.py b/pyanaconda/ui/gui/spokes/lib/passphrase.py
index 1f3ebce..bd41bd5 100644
--- a/pyanaconda/ui/gui/spokes/lib/passphrase.py
+++ b/pyanaconda/ui/gui/spokes/lib/passphrase.py
@@ -141,7 +141,7 @@ class PassphraseDialog(GUIObject):
         if sensitive:
             self._confirm_entry.grab_focus()
 
-        return False
+        return True
 
     def on_confirm_changed(self, entry):
         if not self._save_button.get_sensitive() and \
diff --git a/pyanaconda/ui/gui/spokes/password.glade b/pyanaconda/ui/gui/spokes/password.glade
index 7670f09..75ac957 100644
--- a/pyanaconda/ui/gui/spokes/password.glade
+++ b/pyanaconda/ui/gui/spokes/password.glade
@@ -85,8 +85,6 @@
                         <property name="can_focus">True</property>
                         <property name="visibility">False</property>
                         <property name="invisible_char">●</property>
-                        <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                        <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                       </object>
                       <packing>
                         <property name="left_attach">1</property>
@@ -102,8 +100,6 @@
                         <property name="visibility">False</property>
                         <property name="invisible_char">●</property>
                         <property name="activates_default">True</property>
-                        <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                        <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                       </object>
                       <packing>
                         <property name="left_attach">1</property>
diff --git a/pyanaconda/ui/gui/spokes/source.glade b/pyanaconda/ui/gui/spokes/source.glade
index 2cc41f2..a826145 100644
--- a/pyanaconda/ui/gui/spokes/source.glade
+++ b/pyanaconda/ui/gui/spokes/source.glade
@@ -437,8 +437,6 @@
                         <property name="can_focus">True</property>
                         <property name="visibility">False</property>
                         <property name="invisible_char">●</property>
-                        <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                        <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                       </object>
                       <packing>
                         <property name="left_attach">1</property>
@@ -1151,8 +1149,6 @@
                             <property name="invisible_char">●</property>
                             <property name="invisible_char_set">True</property>
                             <signal name="changed" handler="on_repoProxy_changed" swapped="no"/>
-                            <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                            <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                           </object>
                           <packing>
                             <property name="left_attach">1</property>
diff --git a/pyanaconda/ui/gui/spokes/user.glade b/pyanaconda/ui/gui/spokes/user.glade
index ae4a190..8277369 100644
--- a/pyanaconda/ui/gui/spokes/user.glade
+++ b/pyanaconda/ui/gui/spokes/user.glade
@@ -173,8 +173,6 @@
                         <property name="visibility">False</property>
                         <property name="invisible_char">●</property>
                         <signal name="changed" handler="_checkPassword" swapped="no"/>
-                        <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                        <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                       </object>
                       <packing>
                         <property name="left_attach">1</property>
@@ -190,8 +188,6 @@
                         <property name="visibility">False</property>
                         <property name="invisible_char">●</property>
                         <signal name="changed" handler="_checkPassword" swapped="no"/>
-                        <signal name="focus-in-event" handler="on_password_focused" swapped="no"/>
-                        <signal name="focus-out-event" handler="on_password_unfocused" swapped="no"/>
                       </object>
                       <packing>
                         <property name="left_attach">1</property>
--
cgit v0.9.1
